app/src/main/java/com/dcrandroid/activities/more/PoliteiaActivity.kt

```diff
@@ -102,19 +110,55 @@ class PoliteiaActivity : BaseActivity(), ProposalNotificationListener,
                 SnackBar.showError(this, R.string.not_connected)
                 return@setOnClickListener
             }
-            syncProposals()
+            if(multiWallet!!.politeia.isConnected){
+                multiWallet!!.politeia.stopSync()
+                setSyncButtonState()
+            }else{
+                syncProposals()
+            }
+
         }
 
         refreshAvailableProposalCategories()
+        setSyncButtonState()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        multiWallet!!.politeia.removeNotificationListener(this.javaClass.name)
     }
 
     private fun syncProposals() = GlobalScope.launch(Dispatchers.Default) {
         try {
+            withContext(Dispatchers.Main){
+                tv_sync_label.setText(R.string.stop_sync)
+                sync_icon.animation = rotateAnim
+            }
             SnackBar.showText(this@PoliteiaActivity, R.string.syncing_proposals)
             multiWallet!!.politeia.sync()
         } catch (e: Exception) {
             e.printStackTrace()
-            SnackBar.showError(this@PoliteiaActivity, R.string.error_syncying_proposals)
+            if (e.message == Dcrlibwallet.ErrContextCanceled) {
+                SnackBar.showError(this@PoliteiaActivity, R.string.politeia_sync_stopped)
+            }else{
+                SnackBar.showError(this@PoliteiaActivity, R.string.error_syncying_proposals)
+            }
+            setSyncButtonState()
+        }
+    }
+
+    private fun setSyncButtonState() = GlobalScope.launch(Dispatchers.Main){
+        if(multiWallet!!.politeia.isConnected){
+            tv_sync_label.setText(R.string.stop_sync)
+
+            if(multiWallet!!.politeia.synced()) {
+                sync_icon.animation = null
```

Comment:
Sync doesn't 'complete', what completed is the fetching of all unfetched proposals and dcrlibwallet will continue to check every 30 minutes for proposal updates.

This is why I don't really support that 'Sync' phrase in this part of the app because it's going to be confused with normal syncing.

Reply:
oh i see, yes it's a bit confusing, we might have to find a suitable synonym